From a9498cf03952e9cd17a7ef5fe72f09a0b6b2365f Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:32 -0400
Subject: [ethernet] igb: Fixed Wake On LAN support

Message-id: <1409062359-13063-15-git-send-email-sassmann@redhat.com>
Patchwork-id: 89131
O-Subject: [RHEL7.1 PATCH 14/81] igb: Fixed Wake On LAN support
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>

This patch fixes Wake on LAN being reported as supported on some Ethernet
ports, in contrary to Hardware capability.

Signed-off-by: Akeem G Abodunrin <akeem.g.abodunrin@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 42ce4126d8bc2e128e1f207cf79bb0623fac498f)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index b0f3666..c3143da 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -2062,14 +2062,15 @@ static void igb_get_wol(struct net_device *netdev, struct ethtool_wolinfo *wol)
 {
  struct igb_adapter *adapter = netdev_priv(netdev);
 
- wol->supported = WAKE_UCAST | WAKE_MCAST |
-    WAKE_BCAST | WAKE_MAGIC |
-    WAKE_PHY;
  wol->wolopts = 0;
 
  if (!(adapter->flags & IGB_FLAG_WOL_SUPPORTED))
   return;
 
+ wol->supported = WAKE_UCAST | WAKE_MCAST |
+    WAKE_BCAST | WAKE_MAGIC |
+    WAKE_PHY;
+
  /* apply any specific unsupported masks here */
  switch (adapter->hw.device_id) {
  default:
-- 
1.7.1